packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx

```diff
@@ -116,6 +141,10 @@ export class FocusZone extends BaseComponent<IFocusZoneProps, {}> implements IFo
 
     const Tag = this.props.elementType || 'div';
 
+    // When we're rendering, if we are currently focused, track the index path
+    // so that we can restore focus after render if needed.
+    this._evaluateFocusBeforeRender();
```

Comment:
Those ones [shouldn't have side effects](https://reactjs.org/docs/strict-mode.html#detecting-unexpected-side-effects) either, and they're both [being deprecated](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html).

Reply:
This might be a case for [`getSnapshotBeforeUpdate`](https://reactjs.org/docs/react-component.html#getsnapshotbeforeupdate)?

Reply:
I think this is safe here, and I can elaborate in the comment. We are applying no changes to the DOM here, simply recording if focus is present within the zone before the update, so that we can restore if needed after. It is SSR and node.js safe, as the code changes are careful not to access DOM api surface without testing first (getDocument, null check)

Reply:
It should be fine to call this more than once.